src/mainnet/target-assets/fortETH.sol

```diff
@@ -0,0 +1,462 @@
+    /// @param _receiver - The address of the receiver of shares
+    /// @return _assets - The amount of underlying assets received
+    function mint(uint256 _shares, address _receiver) external override nonReentrant returns (uint256 _assets) {
+        if (_shares >= maxMint(msg.sender)) revert InsufficientDepositCap();
+
+        _assets = previewMint(_shares);
+
+        _deposit(msg.sender, _receiver, _assets, _shares);
+
+        IERC20(address(asset)).safeTransferFrom(msg.sender, address(this), _assets);
+        
+        _distributeAssets(_assets);
+
+        return _assets;
+    }
+
+    /// @dev Distribute buffered WETH between AssetVaults
+    function distributeBufferedAssets(address _receiver, uint256 _minBounty) external nonReentrant {
+        uint256 bufferedBalance = ERC20(WETH).balanceOf(address(this));
+        if (!bufferedBalance>0) revert ZeroAmount();
+
+        uint256 _harvestBounty = _fees.harvestBountyPercentage;
+        if (_harvestBounty > 0) {
+                _harvestBounty = (_harvestBounty * bufferedBalance) / DENOMINATOR;
+                if (!(_harvestBounty >= _minBounty)) revert InsufficientAmountOut();
+                
+                IERC20(WETH).safeTransfer(_receiver, _harvestBounty);
+            }
+         _distributeAssets(bufferedBalance - _harvestBounty);
+    }
+
+    /// @dev Burns shares from owner and sends exact assets of underlying assets to _receiver. If the _owner is whitelisted, no withdrawal fee is applied
+    /// @param _assets - The amount of underlying assets to receive
+    /// @param _receiver - The address of the receiver of underlying assets
+    /// @param _owner - The owner of shares
+    /// @return _shares - The amount of shares burned
+    function withdraw(uint256 _assets, address _receiver, address _owner) external override nonReentrant returns (uint256 _shares) {}
+     
+
+    /// @dev Burns exact shares from owner and sends assets of underlying tokens to _receiver. If the _owner is whitelisted, no withdrawal fee is applied
+    /// @param _shares - The shares to burn
+    /// @param _receiver - The address of the receiver of underlying assets
+    /// @param _owner - The owner of shares to burn
+    /// @return _assets - The amount of assets returned to the user
+    function redeem(uint256 _shares, address _receiver, address _owner) external override nonReentrant returns (uint256 _assets) {}
+       
+    /// @dev Adds emitting of YbTokenTransfer event to the original function
+    function transfer(address to, uint256 amount) public override returns (bool) {
+        balanceOf[msg.sender] -= amount;
+
+        // Cannot overflow because the sum of all user
+        // balances can't exceed the max uint256 value.
+        unchecked {
+            balanceOf[to] += amount;
+        }
+
+        emit Transfer(msg.sender, to, amount);
+        emit YbTokenTransfer(msg.sender, to, amount, convertToAssets(amount));
+        
+        return true;
+    }
```

Comment:
What is the best practice here? Shouldn't there be some method that we inject and the rest of `transfer` remains the standard implementation? The current code looks like everything is copied and one line added for our use case. Not sure this is how we should be doing it.